pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/mercury/MercuryOrderDao.java

```diff
@@ -197,15 +197,66 @@ public int create(MercuryOrderDto mercuryOrderDto) {
         throw new IllegalStateException("Call the version of this method that includes the raw json payload");
     }
 
+    public int create(MercuryOrderDto mercuryOrderDto, String messageJson) {
+        SimpleResult results = inTransaction(conn -> {
+            SimpleResult execResult = new SimpleResult();
+            try (PreparedStatement stmt = conn.prepareStatement(SQL_INSERT_MERCURY_ORDER, Statement.RETURN_GENERATED_KEYS)) {
+                stmt.setString(1, mercuryOrderDto.getOrderId());
+                stmt.setLong(2, mercuryOrderDto.getOrderDate());
+                stmt.setString(3, mercuryOrderDto.getDdpParticipantId());
+                stmt.setInt(4, mercuryOrderDto.getKitTypeId());
+                stmt.setString(5, mercuryOrderDto.getBarcode());
+                stmt.setInt(6, mercuryOrderDto.getDdpInstanceId());
+                stmt.setString(7, mercuryOrderDto.getCreatedBy().orElse(""));
+                stmt.setString(11, messageJson);
+                stmt.setNull(12, Types.VARCHAR);
+                if (mercuryOrderDto.getTissueId() != null) {
+                    stmt.setString(8, String.valueOf(mercuryOrderDto.getTissueId()));
```

Comment:
We're setting this as a string if it's not null, but then if it is null, we set it to null of type integer?  Seems like this should be setInt.

Reply:
Old code, fixed now